Cargo.toml

```diff
@@ -61,6 +61,9 @@ debug = false
 # use LTO for smaller binaries (that take longer to build)
 [profile.release]
 lto = true
+strip = true
+opt-level = "z"
+codegen-units = 1
```

Comment:
```suggestion
strip = true
opt-level = "s"
```

Reply:
Nice job! This is so comprehensive!

I actually care somewhat that we get automatic vectorization (although I'm unsure where it would be applicable in the code), so this is perfect.